src/main/java/com/melon/portfoliomanager/services/HttpService.java

```diff
@@ -0,0 +1,61 @@
+package com.melon.portfoliomanager.services;
+
+import com.melon.portfoliomanager.dtos.responses.StockPricesDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.net.URI;
+
+@Service
+public class HttpService {
+    private static final Logger logger = LoggerFactory.getLogger(HttpService.class);
+    private final WebClient webClient;
+
+    @Autowired
+    public HttpService(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    public ResponseEntity<StockPricesDTO> getStockPrices(URI uri) {
```

Comment:
if we want this class to be more abstract, can't we have something like
`public <T> ResponseEntity<T> execute(URI uri, Class<T> responseType) {`

and we could call it like this
`httpService.execute("http://example.com", StockPricesDTO.class)`

I'm not exactly sure how to cast it though, but it may be worth looking into

Reply:
Yeah, it could be done but the idea for this service is not to provide some general abstraction over sending http requests. (please refer to my comment here: https://github.com/Hypermood/Portfolio-Manager/pull/8#discussion_r1287116022)